### ISSUE TITLE
Read-only Git Repositories

### ISSUE THREAD

**Issue by soulofmischief** [keybase/keybase-issues#3303]
Hi,

I use Keybase's encrypted repositories to manage dotfile configurations.

I would like a way to provide read-only access to some repositories on a per-machine basis because presently it seems any machine can push to the repo, meaning that if my most insecure machine gets compromised then it compromises every machine that uses the same dotfiles, if I pull from origin before noticing some changes were not mine.

**Comment by strib** [comment]
You can make a team, with yourself as a writer and another account you make as a reader.  Then sign into the reader account from the machines you don't trust.  Doesn't that match exactly what you want?

**Comment by soulofmischief** [comment]
Sure, but then I'm handling several accounts, each with their own private paper key. It just doesn't really scale when you're dealing with dozens of machines.

**Comment by junderw** [comment]
> IMO that is broken right now, team permissions are a must and it seems any team member can push to master.

Nope. I can confirm, Readers in our team can not push.

> I would like a way to provide read-only access to some repositories on a per-machine basis

Currently keybase is not set up to deal with machines as separate entities.

The smallest form of permission change is a single user.

A single user can have multiple devices, but Keybase doesn't support permission changes on a per-device basis.

If it did, things would get very complicated very quick.

You can get around this by making one extra user called soulofmischiefreadonly and use that account on all machines you don't trust. then invite that user to a team as a Reader. it will only be able to pull, not push.

You don't need to make a new user per-device. Just make two users and only login to your main account on devices you can trust.

Simple.

**Comment by soulofmischief** [comment]
Yeah, I dug into it more and found that you can set user permissions, it just wasn't immediately obvious when looking at the team section with only one user.

> You can get around this by making one extra user called soulofmischiefreadonly and use that account on all machines you don't trust

That could work, thanks for the suggestion. Currently I'm using the Ansible git module to work around this limitation so I'll see how far that takes me.

> If it did, things would get very complicated very quick.

Just curious, why is that?
